tests/std/tests/P0896R4_ranges_alg_generate_n/test.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+constexpr void smoke_test() {
+    using ranges::generate_n, ranges::equal;
+    {
+        int output[] = {13, 42, 1367};
+        auto result  = generate_n(ranges::begin(output), ranges::distance(output), [calls_to_generate = -1]() mutable {
+            ++calls_to_generate;
+            return calls_to_generate;
+        });
+        for (int i = 0; i < 3; ++i) {
+            assert(i == output[i]);
+        }
+        assert(result == ranges::end(output));
+    }
+    {
+        int expected_output[] = {13, 42, 1367};
+        int output[]          = {13, 42, 1367};
+        auto result           = generate_n(ranges::begin(output), 0, [calls_to_generate = -1]() mutable {
+            ++calls_to_generate;
+            return calls_to_generate;
+        });
+        assert(ranges::equal(output, expected_output));

```

Comment:
Line 13 is `using` `ranges::equal`:
```suggestion
        assert(equal(output, expected_output));
```